exp/runtime/hooks/api/v1alpha1/common_types.go

```diff
@@ -16,6 +16,46 @@ limitations under the License.
 
 package v1alpha1
 
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Response can set and get fields common to all Response types.
+// +kubebuilder:object:generate=false
+type Response interface {
```

Comment:
Q: should this be `ReponseObject` given that it is an object + some behavior that are specific for a ResponseObject?